### PR TITLE
docs: updated nuxt bridge migration guide

### PR DESCRIPTION
**Pull request by hecktarzuli**
added a section about the public runtime config change in Nuxt 3

### 🔗 Linked issue

Resolves #7667 

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Added section to bridge migration guide about public runtime configs in Nuxt 3

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/bridge/1.overview.md [hunk 1]
```diff
@@ -149,6 +149,21 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
 In case you need to extend options provided by `./.nuxt/tsconfig.json` further, you can use the `alias` property withing your `nuxt.config`. `nuxi` will pick them up and extend `./.nuxt/tsconfig.json` accordingly.
 ::
 
+## Update Public Runtime Config References
+Nuxt 3 approaches runtime configs differently than Nuxt 2.  Now, all public runtime config properties are behind a property called `public`.  If you use public runtime configs, you'll need to update your code.
```
Comment (Atinux) [inline docs/content/bridge/1.overview.md:153]:
```suggestion

Nuxt 3 approaches runtime configs differently than Nuxt 2.  Now, all public runtime config properties are behind a property called `public`.  If you use public runtime configs, you'll need to update your code.
```

File: docs/content/bridge/1.overview.md [hunk 2]
```diff
@@ -149,6 +149,22 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
 In case you need to extend options provided by `./.nuxt/tsconfig.json` further, you can use the `alias` property withing your `nuxt.config`. `nuxi` will pick them up and extend `./.nuxt/tsconfig.json` accordingly.
 ::
 
+## Update Public Runtime Config References
+
+Nuxt 3 approaches runtime configs differently than Nuxt 2.  Now, all public runtime config properties are behind a property called `public`.  If you use public runtime configs, you'll need to update your code.
+
+```
```
Comment (Atinux) [inline docs/content/bridge/1.overview.md:156]:
```suggestion
```js
```

File: docs/content/bridge/1.overview.md [hunk 3]
```diff
@@ -149,6 +149,22 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
...
+// nuxt.config.js
+publicRuntimeConfig: {
+  foo: 'bar'
+}
+```
+``` diff
```
Comment (Atinux) [inline docs/content/bridge/1.overview.md:162]:
```suggestion

```diff
```

**Review by Atinux — APPROVED** [review]

File: docs/content/bridge/1.overview.md [hunk 4]
```diff
@@ -149,6 +149,23 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
...
 
+## Update Public Runtime Config References
+
+Nuxt 3 approaches runtime configs differently than Nuxt 2.  Now, all public runtime config properties are behind a property called `public`.  If you use public runtime configs, you'll need to update your code.
+
+```js
+// nuxt.config.js
+publicRuntimeConfig: {
+  foo: 'bar'
+}
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:160]:
```suggestion
- privateRuntimeConfig: {
-   apiKey: process.env.NUXT_API_KEY || 'super-secret-key'
- },
- publicRuntimeConfig: {
-   websiteURL: 'https://public-data.com'
- }
+ runtimeConfig: {
+   apiKey: process.env.NUXT_API_KEY || 'super-secret-key',
+   public: {
+     websiteURL: 'https://public-data.com'
+   }
+ }
```

File: docs/content/bridge/1.overview.md [hunk 5]
```diff
@@ -149,6 +149,23 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
 In case you need to extend options provided by `./.nuxt/tsconfig.json` further, you can use the `alias` property withing your `nuxt.config`. `nuxi` will pick them up and extend `./.nuxt/tsconfig.json` accordingly.
 ::
 
+## Update Public Runtime Config References
+
+Nuxt 3 approaches runtime configs differently than Nuxt 2.  Now, all public runtime config properties are behind a property called `public`.  If you use public runtime configs, you'll need to update your code.
+
+```js
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:156]:
```suggestion
```diff
```

File: docs/content/bridge/1.overview.md [hunk 6]
```diff
@@ -149,6 +149,23 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
 In case you need to extend options provided by `./.nuxt/tsconfig.json` further, you can use the `alias` property withing your `nuxt.config`. `nuxi` will pick them up and extend `./.nuxt/tsconfig.json` accordingly.
 ::
 
+## Update Public Runtime Config References
+
+Nuxt 3 approaches runtime configs differently than Nuxt 2.  Now, all public runtime config properties are behind a property called `public`.  If you use public runtime configs, you'll need to update your code.
+
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:155]:
```suggestion

First, you'll need to combine your `publicRuntimeConfig` and `privateRuntimeConfig` properties into a new one called `runtimeConfig`, with the public config within a key called `public`.

```

File: docs/content/bridge/1.overview.md [hunk 7]
```diff
@@ -149,6 +149,23 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
 In case you need to extend options provided by `./.nuxt/tsconfig.json` further, you can use the `alias` property withing your `nuxt.config`. `nuxi` will pick them up and extend `./.nuxt/tsconfig.json` accordingly.
 ::
 
+## Update Public Runtime Config References
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:152]:
```suggestion
## Update Runtime Config
```

File: docs/content/bridge/1.overview.md [hunk 8]
```diff
@@ -149,6 +149,23 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
...
+
+Nuxt 3 approaches runtime configs differently than Nuxt 2.  Now, all public runtime config properties are behind a property called `public`.  If you use public runtime configs, you'll need to update your code.
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:154]:
```suggestion
Nuxt 3 approaches runtime configs differently than Nuxt 2, using a new combined `runtimeConfig` option.
```

File: docs/content/bridge/1.overview.md [hunk 9]
```diff
@@ -149,6 +149,23 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
...
+
+```js
+// nuxt.config.js
+publicRuntimeConfig: {
+  foo: 'bar'
+}
+```
+
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:162]:
```suggestion

This also means that when you need to access public runtime config, it's behind a property called `public`. If you use public runtime configs, you'll need to update your code.

```

File: docs/content/bridge/1.overview.md [hunk 10]
```diff
@@ -149,6 +149,36 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
...
++   }
++ }
+```
+
+This also means that when you need to access public runtime config, it's behind a property called `public`. If you use public runtime configs, you'll need to update your code.
+
+```diff
+// MyWidget.vue
+- <div>Foo: {{ $config.foo }}</div>
++ <div>Foo: {{ $config.public.foo }}</div>
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:179]:
```suggestion
- <div>Foo: {{ $config.websiteURL }}</div>
+ <div>Foo: {{ $config.public.websiteURL }}</div>
```

File: docs/content/bridge/1.overview.md [hunk 11]
```diff
@@ -149,6 +149,36 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
 In case you need to extend options provided by `./.nuxt/tsconfig.json` further, you can use the `alias` property withing your `nuxt.config`. `nuxi` will pick them up and extend `./.nuxt/tsconfig.json` accordingly.
 ::
 
+## Update Runtime Config
+
+Nuxt 3 approaches runtime configs differently than Nuxt 2, using a new combined `runtimeConfig` option.
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:154]:
```suggestion
Nuxt 3 approaches runtime config differently than Nuxt 2, using a new combined `runtimeConfig` option.
```

File: docs/content/bridge/1.overview.md [hunk 12]
```diff
@@ -149,6 +149,36 @@ Overwriting options such as `"compilerOptions.paths"` with your own configuratio
...
+- }
++ runtimeConfig: {
++   apiKey: process.env.NUXT_API_KEY || 'super-secret-key',
++   public: {
++     websiteURL: 'https://public-data.com'
++   }
++ }
+```
+
+This also means that when you need to access public runtime config, it's behind a property called `public`. If you use public runtime configs, you'll need to update your code.
```
Comment (danielroe) [inline docs/content/bridge/1.overview.md:174]:
```suggestion
This also means that when you need to access public runtime config, it's behind a property called `public`. If you use public runtime config, you'll need to update your code.
```

**Review by danielroe — APPROVED** [review]
Thank you ❤️

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**